### PR TITLE
[VL-130] - Add independent checks about send notification

### PR DESCRIPTION
**Pull request by kin0992**
Add first part of checks related to the send notification use case

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by datalek — COMMENTED** [review]
[Here](https://github.com/pagopa/pn-local-emulator/pull/63) some optional suggestions, you can close it or merge it

File: src/domain/NewNotificationRepository.ts [hunk 1]
```diff
@@ -14,6 +15,58 @@ export type NewNotificationRecord = {
...
+  pipe(
+    record.input.body.recipients,
+    RA.every((recipient) =>
+      pipe(
+        recipient.payment,
+        O.fromNullable,
+        O.map((_) => _.creditorTaxId),
+        O.isSome
+      )
+    )
```
Comment (datalek) [inline src/domain/NewNotificationRepository.ts:54]: What do you think about this suggestion? Feel free to keep it or close it

```suggestion
    RA.every(({ payment }) => pipe(payment?.creditorTaxId, O.fromNullable, O.isSome))
```

File: src/domain/NewNotificationRepository.ts [hunk 2]
```diff
@@ -14,6 +15,58 @@ export type NewNotificationRecord = {
...
+  pipe(
+    record.input.body.recipients,
+    RA.every((recipient) =>
+      pipe(
+        recipient.payment,
+        O.fromNullable,
+        O.map((_) => _.noticeCode),
+        O.isSome
+      )
+    )
```
Comment (datalek) [inline src/domain/NewNotificationRepository.ts:67]: Feel free to keep it or close it
```suggestion
    RA.every(({ payment }) => pipe(payment?.noticeCode, O.fromNullable, O.isSome))
```

File: src/domain/NewNotificationRepository.ts (hunk 1 above)
Comment (kin0992) [reply src/domain/NewNotificationRepository.ts]: Nice! Done! 🍻

File: src/domain/NewNotificationRepository.ts (hunk 2 above)
Comment (kin0992) [reply src/domain/NewNotificationRepository.ts]: Done!

**Review by datalek — APPROVED** [review]
💥

**Pull request merged**